keras_eval/eval.py

```diff
@@ -145,6 +149,32 @@ def evaluate(self, data_dir=None, top_k=1, filter_indices=None, confusion_matrix
 
         return self.probabilities, self.labels
 
+    def evaluate_combination_concepts(self):
+        '''
+        Sets probabilities if model concepts are to be combined in test
+        '''
+
+        if utils.compare_group_test_concepts(self.concept_labels, self.concept_dictionary) and utils.check_concept_unique(self.concept_dictionary):
+            for clas in self.concept_dictionary:
+
+                if clas['group'] in self.group_id_dict.keys():
+                    self.group_id_dict[clas['group']].append(clas['class_index'])
+                else:
+                    self.group_id_dict[clas['group']] = [clas['class_index']]
+
+            self.combined_probs = [[[0.0, ] * len(self.concept_labels)] * len(self.probabilities[0])]
+            self.combined_probs = np.array(self.combined_probs)
+            id = 0
+            for concept_label in self.concept_labels:
```

Comment:
You can avoid incrementing the `id` by doing:
```
for idx, concept_label in enumerate(self.concept_labels):
```

and `idx` will increment for each `self.concept_labels` item.